`fscan` panic while port scanning

### Describe the bug
As titled, package `fscan` panic while port scanning.
release `1.8.3`, [binary](https://github.com/shadow1ng/fscan/releases/download/1.8.3/fscan) downloaded from the official repo releases has **no issue**.

1 < people reproduced this issue.

### Steps To Reproduce
```
> sudo fscan -h 10.234.0.1/16 -p 7810 -no

   ___                              _    
  / _ \     ___  ___ _ __ __ _  ___| | __ 
 / /_\/____/ __|/ __| '__/ _` |/ __| |/ /
/ /_\\_____\__ \ (__| | | (_| | (__|   <    
\____/     |___/\___|_|  \__,_|\___|_|\_\   
                     fscan version: 1.8.3
start infoscan
brbrbr is alive

[*] Icmp alive hosts len is: 55
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 98 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc00048a480?, 0xc0000f7560?}, 0x10?}, 0x10?, 0x0?, 0x0?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
```

And (sometimes different ?)
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 86 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc00011e0a0?, 0x111a570?}, 0xc0003fe5d0?}, 0x111a570?, 0xc000990c90?, 0x1117040?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 88 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc000147630?, 0x111a570?}, 0xc000991fb0?}, 0x1117040?, 0xc000987e00?, 0xd66c8b?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 39 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc0003b83d0?, 0x0?}, 0x0?}, 0x0?, 0x0?, 0x0?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 170 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc000166140?, 0x0?}, 0x0?}, 0x0?, 0x0?, 0x1003031cec60008?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 437 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc0004c2150?, 0x1116ee0?}, 0x1ae1568?}, 0xd67765?, 0x6?, 0x1116ee0?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 191 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc00013c080?, 0x1116ee0?}, 0x1ae1568?}, 0xd67765?, 0x6?, 0x1116ee0?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 89 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc00013c0b0?, 0x0?}, 0x0?}, 0xc0004bc540?, 0x10?, 0x10?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbb5525]

goroutine 263 [running]:
github.com/shadow1ng/fscan/Plugins.PortConnect({{0xc0003b8480?, 0x1116ee0?}, 0x1ae1568?}, 0xd67765?, 0x6?, 0x1116ee0?)
	github.com/shadow1ng/fscan/Plugins/portscan.go:77 +0x105
github.com/shadow1ng/fscan/Plugins.PortScan.func2()
	github.com/shadow1ng/fscan/Plugins/portscan.go:55 +0x79
created by github.com/shadow1ng/fscan/Plugins.PortScan in goroutine 1
	github.com/shadow1ng/fscan/Plugins/portscan.go:53 +0x3a6
```

### Expected behavior

works.

### Additional

Try compiling with `buildGo119Module` and panic the same.

### Notify maintainers
@Misaka13514 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.12, NixOS, 24.05 (Uakari), 24.05.20231127.8cfef69`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/security/fscan/default.nix
+++ b/pkgs/tools/security/fscan/default.nix
@@ -2,13 +2,13 @@
 
 buildGoModule rec {
   pname = "fscan";
-  version = "1.8.3";
+  version = "1.8.3-build3";
 
   src = fetchFromGitHub {
     owner = "shadow1ng";
     repo = "fscan";
     rev = version;
-    hash = "sha256-uoM/nMtgIqyzpOoSQKD5k4LXAXoA8G5N4In8tZlngqs=";
+    hash = "sha256-GtOCd8JaR6tx8hoB+P9QXrEnN7Wvmv7jddhc2/8hjvQ=";
   };
 
   vendorHash = "sha256-hvb2IfypwYauF3ubE36u0bTU+l/FWP/CZt6dFd9zc6s=";
@@ -18,7 +18,6 @@ buildGoModule rec {
     homepage = "https://github.com/shadow1ng/fscan";
     license = licenses.mit;
     maintainers = with maintainers; [ Misaka13514 ];
-    platforms = with platforms; unix ++ windows;
     mainProgram = "fscan";
   };
 }

